### PR TITLE
cast ints from/to bool

### PR DESCRIPTION
**Pull request by kali**

### CODE REVIEW

File: core/src/datum.rs [hunk 1]
```diff
@@ -320,6 +320,27 @@ impl TryInto<f64> for bool {
     }
 }
 
+macro_rules! cast_to_and_from_bool {
+    ($t: ty) => {
+        impl TryInto<bool> for $t {
+            fn try_into(&self) -> TractResult<bool> {
+                Ok(*self == 0)
```
Comment (bio43) [inline core/src/datum.rs:327]: The int -> bool cast is not well defined, but according to https://github.com/onnx/onnx/issues/2285 , it seems that truncating the leading bits is the right approach. Suggestion:
```suggestion
                Ok((*self & 1) == 1)
```

Comment (kali) [reply core/src/datum.rs:327]: wow. I was more tired than I thought.

**Comment by kali** [comment]
I'll stick to != 0 though. That's what MS onnxruntime does, and it it feels more natural to me than truncation. Probably matter of taste though.

**Comment by bio43** [comment]
@kali ok, I tested this branch and can confirm it works for my use case.

**Pull request merged**